### PR TITLE
fix(#168): passing `request` object in template context

### LINKED ISSUE #168: link to redirect unauthenticated user for comment, reaction, etc. doesn't work
Screenshot (42)_LI
Screenshot (44)_LI
Screenshot (46)

### Environment
* python version 3.8
* django Version 3.1
* django-comments-dab 2.5

### Describe the bug
Note: I uninstalled the app from the packages section and put it in my directory Apps

### Steps To Reproduce
1.
2.
3.

### PR DESCRIPTION
**Pull request by ZendaInnocent**
Set login url next parameter to absolute url of the model object.

fixes #168

### CODE REVIEW

**Review by abhiabhi94 — CHANGES_REQUESTED** [review]
Hey @ZendaInnocent, this will not fix the actual problem, it just fixes this part of the template. There are other parts that need the `request` object.

If you really want to have a stab at this, you will have to add the `request` object to the `context` dictionary here:

https://github.com/Radi85/Comment/blob/0d7bc4b99fa5573fe4bbda9ab8f32099ed8f4ffe/comment/templatetags/comment_tags.py#L74-L82
`request` is already available in this function.

Also, you will probably need to change the current tests, I think

**Comment by ZendaInnocent** [comment]
Adding `request` object in `DABContext.__call__` method won't be sufficient?

I tried to add it there and seems to work fine.

**Comment by abhiabhi94** [comment]
> Adding `request` object in `DABContext.__call__` method won't be sufficient?
> 
> I tried to add it there and seems to work fine.

The `DABContext` call is [used in most of the views](https://github.com/Radi85/Comment/blob/develop/comment/mixins.py#L41) when rendering a template.
When rendering a template using the method `rendering_to_string`, [a lot of metadata is passed to the template by Django](https://docs.djangoproject.com/en/dev/topics/templates/#django.template.loader.render_to_string). `request` is one of them.

So, my point is request is already available for other templates that are rendered through views. It is not available in this case, because we are using `inclusion_tag` to render the template, and it was not passed explicitly by it.

In short, adding `request` to `DABContext` seems more of an overkill to me, just adding it to the `context` dictionary used by `render_comments` template tag should be sufficient in this case.

**Comment by ZendaInnocent** [comment]
Thanks for the explanations.

File: comment/tests/test_template_tags.py [hunk 1]
```diff
@@ -74,6 +75,8 @@ def test_render_comments(self):
         # no pagination
         self.assertEqual(data['comments'].count(), count)  # parent comment only
         self.assertEqual(data['login_url'], settings.LOGIN_URL)
+        # check if `request` object is passed in context template
+        self.assertIsInstance(data['request'], WSGIRequest)
```
Comment (abhiabhi94) [inline comment/tests/test_template_tags.py:79]: perhaps this should be:
```suggestion
        self.assertEqual(data['request'], request)
```

**Review by abhiabhi94 — APPROVED** [review]
Thanks!

**Comment by Radi85** [comment]
@ZeroCoolHacker @abhiabhi94 
I see that @ZendaInnocent first solution is more eligible. The request object is only needed here to get the path value which is already available for us with `model_object.get_absolute_url`.

**Comment by Radi85** [comment]
PS: First solution was replacing `request.path` with `model_object.get_absolute_url` in the template and it is sufficient.

**Comment by abhiabhi94** [comment]
> PS: First solution was replacing request.path with model_object.get_absolute_url in the template and it is sufficient.

Actually the `request` object is also used to redirect after login for other things like :+1: , unlike, flag, subscribe etc, so adding `request` would require lesser work.

Also, `model_obj.absolute_url` would not wouldn't always be correct especially when the comment is not on the first page. 
e.g `/post?page=2`

**Review by Radi85 — APPROVED** [review]

**Comment by Radi85** [comment]
Thanks @ZendaInnocent for your contribution!

**Comment by abhiabhi94** [comment]
I think the commit message should have been something similar to the one that is the title of the issue closed at the moment. This will be potentially more helpful in generating the `changelog` and also puts the change in the perspective of the end user.
Although, I am the one to blame, I edited it as a lesser appropriate one earlier.

**Comment by ZendaInnocent** [comment]
> I think the commit message should have been something similar to the one that is the title of the issue closed at the moment. This will be potentially more helpful in generating the `changelog` and also puts the change in the perspective of the end user.
> Although, I am the one to blame, I edited it as a lesser appropriate one earlier.

Amending the commit and push to the branch will work?

**Comment by abhiabhi94** [comment]
> Amending the commit and push to the branch will work?

No issues, it was just a reminder for the future.

**Comment by ZendaInnocent** [comment]
> > Amending the commit and push to the branch will work?
> 
> No issues, it was just a reminder for the future.

Okay, thanks.
